qualityattenuation/qualityattenuation.go

```diff
@@ -0,0 +1,133 @@
+// Implements a data structure for quality attenuation.
+
+package qualityattenuation
+
+import (
+	"math"
+
+	"github.com/influxdata/tdigest"
+)
+
+type SimpleQualityAttenuation struct {
+	EmpiricalDistribution     *tdigest.TDigest
+	Offset                    float64
+	Offset_sum                float64
+	Offset_sum_of_squares     float64
+	Number_of_samples         int64
+	Number_of_losses          int64
+	Latency_eq_loss_threshold float64
+	Minimum_latency           float64
+	Maximum_latency           float64
+}
+
+func NewSimpleQualityAttenuation() *SimpleQualityAttenuation {
+	return &SimpleQualityAttenuation{
+		EmpiricalDistribution:     tdigest.NewWithCompression(50),
+		Offset:                    0.1,
+		Offset_sum:                0.0,
+		Offset_sum_of_squares:     0.0,
+		Number_of_samples:         0,
+		Number_of_losses:          0,
+		Latency_eq_loss_threshold: 15.0, // Count latency greater than this value as a loss.
+		Minimum_latency:           0.0,
+		Maximum_latency:           0.0,
+	}
+}
+
+func (qa *SimpleQualityAttenuation) AddSample(sample float64) {
+	if sample <= 0.0 {
+		// Ignore zero or negative samples because they cannot be valid.
+		// TODO: This should raise a warning and/or trigger error handling.
```

Comment:
Is this case something that the caller would be able to do something about? If it's not, then we can just eat up the error silently and move on. Otherwise, we could make the method return an `error`. If the function succeeds, we return `nil` and otherwise return the result of `fmt.Errorf` (or something like that) which the caller can detect and change their behavior. Again, though, if the caller cannot change their behavior and would simply accept that the datapoint could not be added, perhaps there is no reason to return anything? 

On second thought, even if the caller can only log that an error occurred, perhaps returning an error is a "good idea"? 

I am curious what you think!

Reply:
I added the return value as you proposed.  I don't have a very strong opinion either way on this one, but it seems useful that the caller can be  notified when a sample is dropped.